internal/bufferedwrites/upload_handler_test.go

```diff
@@ -340,3 +344,54 @@ func (t *UploadHandlerTest) TestCreateObjectChunkWriterIsCalledWithCorrectReques
 	err = t.uh.Upload(b)
 	require.NoError(t.T(), err)
 }
+
+func (t *UploadHandlerTest) TestDestroy() {
+	testCases := []struct {
+		name           string
+		uploadChClosed bool
+	}{
+		{
+			name:           "UploadChNotClosed",
+			uploadChClosed: false,
+		},
+		{
+			name:           "UploadChClosed",
+			uploadChClosed: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func() {
+			// Create some blocks.
+			var blocks []block.Block
+			for i := 0; i < 5; i++ {
+				b, err := t.blockPool.Get()
+				require.NoError(t.T(), err)
+				blocks = append(blocks, b)
+			}
+			// CreateObjectChunkWriter -- should be called once.
+			writer := &storagemock.Writer{}
+			t.mockBucket.On("CreateObjectChunkWriter", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(writer, nil)
+			// Upload the blocks.
+			for _, b := range blocks {
+				err := t.uh.Upload(b)
```

Comment:
Instead of calling uploader go routine, shall we manually add to uploadCh and increment wait group. That way we will be sure that the buffers are  being cleared via Destroy